### PR TITLE
Allow parsing JavaScript proto definitions from JSON

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR adds the new `spine-proto-js-plugin` which generates the `fromJson(json)` method for the JS proto messages.

The `fromJson` method allows to parse the message from the JSON string.

## Usage

To use the plugin, add
```
classpath "io.spine.tools:spine-proto-js-plugin:$spineBaseVersion"
```
to the `buildscript` section of the Gradle configuration, and then add
```
apply plugin: 'io.spine.tools.proto-js-plugin'
```
to the module compiling `proto` files to JavaScript.

## Needed Project Config

The plugin currently relies on the set of hard-coded settings in the Gradle project configuration. 

The full list of needed settings can be found in the `ProtoJsPlugin` class doc but in general they mimic how the `Spine Web` compiles `proto`s to JS.

Thus, the `Spine Web` is almost ready to use the plugin, except it needs to enable the descriptor set file generation via 
```
task.generateDescriptorSet = true
task.descriptorSetOptions.path = "${projectDir}/build/descriptors/${task.sourceSet.name}/known_types.desc"

```
specified in `protobuf` section of `js.gradle`.

In future the plugin should be able to setup the necessary project config by itself.

## Windows build

Also, the PR contains minor build script changes that should've repaired the Windows build which was previously broken.

## 
Spine version advances to `0.10.91-SNAPSHOT`.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

File: tools/proto-js-plugin/src/main/java/io/spine/code/js/JsFileWriter.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.code.js;
```
Comment (armiol) [inline tools/proto-js-plugin/src/main/java/io/spine/code/js/JsFileWriter.java]: I think the root package should be `io.spine.js`.

Let's discuss the rest of packages in person.

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Pull request merged**